webhooks/pod_webhook.go

```diff
@@ -206,63 +224,78 @@ func (m *PodMutator) getFeatureFlag(ctx context.Context, name string, namespace
 	return ffConfig
 }
 
-func (m *PodMutator) injectSidecar(pod *corev1.Pod, configMap string, featureFlag *corev1alpha1.FeatureFlagConfiguration) ([]byte, error) {
+func (m *PodMutator) injectSidecar(pod *corev1.Pod, featureFlags []*corev1alpha1.FeatureFlagConfiguration) ([]byte, error) {
 	m.Log.V(1).Info(fmt.Sprintf("Creating sidecar for pod %s/%s", pod.Namespace, pod.Name))
-
 	commandSequence := []string{
-		"start", "--uri", "/etc/flagd/config.json",
+		"start",
 	}
+	var envs []corev1.EnvVar
+	var volumeMounts []corev1.VolumeMount
 
-	// FlagD is the default provider name externally
-	if featureFlag.Spec.ServiceProvider != nil && featureFlag.Spec.ServiceProvider.Name != "flagd" {
-		commandSequence = append(commandSequence, "--service-provider")
-		commandSequence = append(commandSequence, "http")
-	}
+	for _, featureFlag := range featureFlags {
+		if featureFlag.Spec.FlagDSpec != nil {
+			if featureFlag.Spec.FlagDSpec.MetricsPort != 0 {
+				flagdMetricsPort = featureFlag.Spec.FlagDSpec.MetricsPort
+			}
+			envs = append(envs, featureFlag.Spec.FlagDSpec.Envs...)
+		}
+		// if remote is explicitly set
+		if featureFlag.Spec.SyncProvider != nil && featureFlag.Spec.SyncProvider.IsRemote() {
```

Comment:
it may be simpler to check `featureFlag.Spec.SyncProvider != nil` just once and use nested `if`s, but I'll leave that up to you.

Reply:
opted for a switch instead